restapi/stories.go

```diff
@@ -58,6 +58,42 @@ func (api api) ReturnStoryByID(w http.ResponseWriter, r *http.Request) render.Re
 	return rest.JSONStatusOK(story)
 }
 
+func (api api) DeleteStoryByID(w http.ResponseWriter, r *http.Request) render.Renderer {
+	// TODO: Delete the image from S3 properly
+	var story models.Story
+	id := chi.URLParam(r, "storyID")
+
+	err := api.database.Preload("Author").Where("id=?", id).First(&story).Error
+	if err != nil {
+		if err == gorm.ErrRecordNotFound {
+			return rest.ErrNotFound(fmt.Sprintf("Could not find story with ID %s", id))
+		}
+		return rest.ErrInternal(api.logger, err)
+	}
+
+	var stories []models.Story
+	err = api.database.Where("author_first_name=?", story.AuthorFirstName).Where("author_last_name=?", story.AuthorLastName).Where("author_country=?", story.AuthorCountry).Find(&stories).Error
```

Comment:
Since you're only doing a count here lets use the gorm `.count()` function: https://stackoverflow.com/questions/60378902/how-get-gorm-query-count-result